File: src/main/java/net/szymonsawicki/reactivetimesheetapp/domain/user/dto/CreateUserDto.java
package net.szymonsawicki.reactivetimesheetapp.domain.user.dto;

import net.szymonsawicki.reactivetimesheetapp.domain.user.User;
import net.szymonsawicki.reactivetimesheetapp.domain.user.type.Role;

public record CreateUserDto(String username, String password, String passwordConfirmation, Role role, String teamId) {
    User toUser() {
        return User.builder()
                .username(username)
                .password(password)
                .role(role)
                .teamId(teamId)
                .build();
    }
}

File: frontend/src/model/CreateUserDto.js
export default class CreateUserDto {
    constructor(username, password, passwordConfirmation, role, teamName) {
        this.username = username;
        this.password = password;
        this.passwordConfirmation = passwordConfirmation;
        this.role = role;
        this.teamName = teamName;
    }
}